src/common/post.ts

```diff
@@ -252,3 +257,68 @@ export const validatePost = (
 export const submitArticleThreshold = parseInt(
   process.env.SUBMIT_ARTICLE_THRESHOLD,
 );
+
+export const insertCodeSnippets = async ({
+  entityManager,
+  post,
+  codeSnippetsUrl,
+}: {
+  entityManager: EntityManager;
+  post: Pick<Post, 'id'>;
+  codeSnippetsUrl: string | undefined;
+}) => {
+  if (!codeSnippetsUrl) {
+    return;
+  }
+
+  try {
+    const codeSnippetsJson = await downloadJsonFile<PostCodeSnippetJsonFile>({
+      url: codeSnippetsUrl,
+    });
+
+    const codeSnippets = codeSnippetsJson.snippets.reduce(
+      (acc, codeSnippetContent, index) => {
+        const checksum = createHash('sha1');
+        checksum.update(codeSnippetContent);
+        const contentHash = checksum.digest('hex');
```

Comment:
We use snippet content hash to uniquely identify each snippet. This is because snippets order and presence in article can change at any time when post is updated (eg. author changes or removes snippets). I use `sha1` because of speed and I think we don't need sha256 for security.